### PR TITLE
271 set trigger status

### PR DESCRIPTION
**Pull request by jonsberg**

### CODE REVIEW

File: app/fn-guru/src/main/frontend/src/pages/app/exec-list/components/hook.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                    exec.trigger = {
+                        id: execTrigger.id,
+                        status: execTrigger.status
+                    }
+                }
+            });
+            setExecsWithTriggerStatus(copy)
+        }
+        setLoading(false)
+    }, [triggerList, execList]);
```
Comment (dominique-dymke) [inline app/fn-guru/src/main/frontend/src/pages/app/exec-list/components/hook.ts:37]: I don't think that's the implementation we need for this - we should add this to the backend which trigger triggered an execution.

Benefits of doing so - we save 1 http call - there can be multiple trigger for the same function --> your implementation does not take care of this

Comment (dominique-dymke) [reply app/fn-guru/src/main/frontend/src/pages/app/exec-list/components/hook.ts:37]: We already have the invocation class - we just need to add the trigger id to the invocation - the moment we rendering it to ApiExec... we fetch the current trigger status and return it in the exec model

File: app/fn-guru/src/main/frontend/src/pages/app/exec-list/components/row-actions.tsx [hunk 2]
```diff
@@ -28,7 +36,28 @@ export default function ({row}: Props) {
...
 }
+
+
+type TriggerSwitchProps = { triggerId: string, status: string }
+const TriggerSwitch: FC<TriggerSwitchProps> = ({triggerId, status}) => {
+    const [setTriggerStatus, setTriggerRequested, loading, error] = useSetTriggerStatus()
+    const current = status === "Active"
+
+    function handleClick(e) {
+        e.stopPropagation()
```
Comment (dominique-dymke) [inline app/fn-guru/src/main/frontend/src/pages/app/exec-list/components/row-actions.tsx:52]: Why do we need this?

File: app/fn-guru/src/main/frontend/src/pages/app/exec-list/components/row-actions.tsx [hunk 3]
```diff
@@ -28,7 +36,28 @@ export default function ({row}: Props) {
...
             </DropdownMenuContent>
         </DropdownMenu>
     )
 }
+
+
+type TriggerSwitchProps = { triggerId: string, status: string }
+const TriggerSwitch: FC<TriggerSwitchProps> = ({triggerId, status}) => {
+    const [setTriggerStatus, setTriggerRequested, loading, error] = useSetTriggerStatus()
+    const current = status === "Active"
```
Comment (dominique-dymke) [inline app/fn-guru/src/main/frontend/src/pages/app/exec-list/components/row-actions.tsx:49]: current is a pretty bad name for a status variable - and you don't really leverage it.

Maybe do something like

```typescript
type TriggerStatus = 'Active' | 'Inactive'

```


```typescript
        <DropdownMenuItem onClick={handleClick}>
            {status == 'Inactive' ? "Activate" : "Deactivate" }
        </DropdownMenuItem>
```

File: app/fn-guru/src/main/frontend/src/pages/app/exec-list/components/row-actions.tsx (hunk 2 above)
Comment (jonsberg) [reply app/fn-guru/src/main/frontend/src/pages/app/exec-list/components/row-actions.tsx]: The row itself had a listener that forwarded to the exec-detail page, the view button in the dropdown already provides this. I removed the row listener and the stopPropagation.

**Pull request merged**